clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -158,6 +161,39 @@ public void testCloseAll() {
         }
     }
 
+    @Test
+    public void testReadFully() throws IOException {
+        RandomAccessFile raf = new RandomAccessFile(TestUtils.tempFile(), "rw");
```

Comment:
The issue with using a real file is that you can't test the scenario that we are trying to fix: `FileChannel.read` returns before the buffer is full.